src/controls/qml/LayerStack.qml

```diff
@@ -56,11 +57,27 @@ Item {
         params["x"] = 0
         params["y"] = 0
         if(typeof firstPage != 'undefined' && firstPage.status === Component.Ready) {
-            var itm=firstPage.createObject(content, params)
+            itm=firstPage.createObject(content, params)
             itm.clip = true
         }
     }
 
+    onFirstPageChanged: {
+        if(typeof itm != 'undefined') {
+            itm.destroy()
+            var params = {}
+            params["width"] = Qt.binding(function() { return width })
+            params["height"] =  Qt.binding(function() { return height })
+            params["x"] = 0
+            params["y"] = 0
+            itm=firstPage.createObject(content, params)
+            itm.clip = true
+            layersChanged()
+        } else {
+            console.log("LayerStack: firstpage has been updated with a null value")
```

Comment:
I certainly am not a javascript export but I think that null and undefined are two different things so this somewhat wrong. Also it says that firstPage has been updated to a null value but the condition that triggers this is if itm (the previously instantiated firstPage object has an undefined type) so in practice this would rather mean that firstPage used to have a ~null type but got updated to something else and we only notice now.

To be fair I would just remove this condition altogether and operate on itm as if firstPage is valid. Javascript doesn't have offer much type guarantees anyway and this feels like https://i.imgflip.com/6om4do.jpg

Reply:
I see now that `Component.onCompleted` does a ~similar check with `typeof firstPage != 'undefined'` but notice that the check is done on `firstPage` not `itm` and that's to avoid a crash if the user just doesn't set `firstPage` which can indeed happen.

So yeah, if you'd like you can keep the same check but make sure you check `firstPage` and not `itm` :)

Reply:
yeah I realised this was weird during the development process, and seeing similar code was why I went ahead with doing it this way - I'll fix that.